internal/ansible/dispatcher/dispatcher_events.go

```diff
@@ -0,0 +1,411 @@
+			TaskUuid:     &tasks.Id,
+			Task:         &tasks.Name,
+			TaskAction:   &hosts.Action,
+			Res: &message.AnsibleRunnerJobEventYamlEventDataRes{
+				Changed: &hosts.Changed,
+			},
+			Uuid: &tasks.Id,
+		},
+		Uuid: tasks.Id,
+	}
+	return &message
+}
+
+func (d *AnsibleDispatcher) RunnerOnFailed(
+	runnerID string,
+	playbookFilename string,
+	playbookUUID string,
+	msgCounter int,
+	host *string,
+	tasks *results.AnsiblePlaybookJSONResultsPlayTaskItem,
+	hosts *results.AnsiblePlaybookJSONResultsPlayTaskHostsItem) *message.AnsibleRunnerJobEventYaml {
+	message := message.AnsibleRunnerJobEventYaml{
+		Event:       "v2_runner_on_failed",
+		RunnerIdent: &runnerID,
+		// StartLine:   0, //Not available
+		// EndLine:     0, //Not available
+		Created: &tasks.Duration.Start,
+		Counter: msgCounter,
+		Stdout:  hosts.Stdout,
+		Stderr:  hosts.Stderr,
+		EventData: &message.AnsibleRunnerJobEventYamlEventData{
+			Playbook:     &playbookFilename,
+			PlaybookUuid: &playbookUUID,
+			Host:         host,
+			TaskUuid:     &tasks.Id,
+			Task:         &tasks.Name,
+			TaskAction:   &hosts.Action,
+			Res: &message.AnsibleRunnerJobEventYamlEventDataRes{
+				Changed: &hosts.Changed,
+			},
+			Uuid: &tasks.Id,
+		},
+		Uuid: tasks.Id,
+	}
+	return &message
+
+}
+
+func (d *AnsibleDispatcher) RunnerOnSkipped(
+	runnerID string,
+	playbookFilename string,
+	playbookUUID string,
+	msgCounter int,
+	host *string,
+	tasks *results.AnsiblePlaybookJSONResultsPlayTaskItem,
+	hosts *results.AnsiblePlaybookJSONResultsPlayTaskHostsItem) *message.AnsibleRunnerJobEventYaml {
+	message := message.AnsibleRunnerJobEventYaml{
+		Event:       "v2_runner_on_skipped",
+		RunnerIdent: &runnerID,
+		// StartLine:   0, //Not available
+		// EndLine:     0, //Not available
+		Created: &tasks.Duration.Start,
+		Counter: msgCounter,
+		Stdout:  hosts.Stdout,
+		Stderr:  hosts.Stderr,
+		EventData: &message.AnsibleRunnerJobEventYamlEventData{
+			Playbook:     &playbookFilename,
+			PlaybookUuid: &playbookUUID,
+			Host:         host,
+			TaskUuid:     &tasks.Id,
+			Task:         &tasks.Name,
+			TaskAction:   &hosts.Action,
+			Res: &message.AnsibleRunnerJobEventYamlEventDataRes{
+				Changed: &hosts.Changed,
+			},
+			Uuid: &tasks.Id,
+		},
+		Uuid: tasks.Id,
+	}
+	return &message
+}
+
+// https://github.com/ansible/ansible-runner/blob/7e60c1f61ee6118f0b943e7897dc3a58ecc043bd/ansible_runner/display_callback/callback/awx_display.py#L530
+func (d *AnsibleDispatcher) PlaybookOnTaskStart(
+	runnerID string,
+	playbookFilename string,
+	playbookUUID string,
+	msgCounter int,
+	host *string,
+	tasks *results.AnsiblePlaybookJSONResultsPlayTaskItem,
+	hosts *results.AnsiblePlaybookJSONResultsPlayTaskHostsItem,
+	taskUUIDs map[string]int,
+	duplicateTaskCount map[string]int) *message.AnsibleRunnerJobEventYaml {
+	taskUUID := tasks.Id
+	if occurence, found := taskUUIDs[taskUUID]; found {
+		if occurence > 1 {
+			// 	 When this task UUID repeats, it means the play is using the
```

Comment:
Don't we always run on localhost so only with one host? It's fine to have this functionality implemented, but I think it won't ever be used in our case.